lib/application.ts

```diff
@@ -683,10 +689,11 @@ export class SNApplication {
     return !isNullOrUndefined(this.getUser()) || this.hasPasscode();
   }
 
-  /**
-   * @returns An array of errors, if any.
-   */
-  public async upgradeProtocolVersion() {
+  public async upgradeProtocolVersion(): Promise<{
+    success?: true,
```

Comment:
Wait why does this default to true?

Reply:
this isn't a default, it's the `true` type. So this says that if `success` is present, it's going to be `true`. an alternative would have been to have the success property be mandatory, and a boolean. But it just resulted in noise wherever we would `return { ... }`.

Reply:
Ah interesting. So TS would complain if you tried to return `{success: false}`?

Reply:
Yes, exactly!